rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,606 @@
+
+#### Platform Specific Implementation <!-- omit in toc -->
+
+**Web:**
+
+There are various third party library implementations for roving tabindex pattern.
+
+We tried out 3 of them to understand the pros and cons for each and decided on X.
+
+**FluentUI** - [Demo](https://codesandbox.io/s/blade-comp-fluentui-focuszone-0sjyv1?file=/src/App.tsx) | [Doc](https://developer.microsoft.com/en-us/fluentui#/controls/web/focuszone)
+
+<details>
+  <summary>Pros and cons</summary>
+
+**Bundle size:** ~12kb
+
+**Pros:**
+
+- Easy to use
+- Handles roving index by default
+- Can handle nested focus zones
+- Provides tabbable focus, means we can enable use of tab + arrow keys
+- Support layout grid behavior (we might not need this)
+
+**Cons:**
+
+- Not much flexibility to write our own logic, doesn’t provide any focus manager
+- Doesn’t provide any other utilities like focus trap, we need to install another package for it `@fluentui/react/lib/focusZoneTrap`
+- FluentUI’s architecture is very complex & confusing because of their need to be cross platform, If we pick this up we might not be able to use their other components to our advantage. Plus their components are not headless.
+
+</details>
+
+**React Aria** - [Demo](https://codesandbox.io/s/blade-comp-react-aria-focus-2v9wtp?file=/src/App.tsx) | [Doc](https://react-spectrum.adobe.com/react-aria/FocusScope.html)
+
+<details>
+  <summary>Pros and cons</summary>
+
+**Bundle size:** ~12kb
+
+ReactAria’s behavior is very different from fluentui, it’s FocusScope doesn’t provide roving index behavior instead it’s like a focus trap. Which is equivalent to FluentUI’s FocusTrapZone
+
+Although ReactAria does provides low level primitives to manage focus like `useFocusManager`
+
+**Pros:**
+
+- Provides other helpful features with it’s focus package, which makes it worth the bundle size compared to other solutions, with other libs we will have to look for alternatives for:
+  - [FocusScope](https://react-spectrum.adobe.com/react-aria/FocusRing.html) (Trapping focus)
+  - [FocusRing](https://react-spectrum.adobe.com/react-aria/FocusRing.html)
+  - [useFocusRing](https://react-spectrum.adobe.com/react-aria/useFocusRing.html)
+- Provides low level primitives for building roving index like behaviors, eg: useFocusManager, getFocusableTreeWalker, createFocusManager
+- Picking react aria will also encourage use of react-native-aria & the eco system of react aria.
+
+**Cons:**
+
+- Doesn’t provide roving index behavior by default
+- A lot of manual logic and code has to be written to implement a proper Roving index component which handles edge cases,
+- The implementation in the codesandbox has various bugs & edge cases, for eg: initially the elements don't have tabIndex set, these kinds of edge cases we need to solve for.
+
+</details>
+
+**Ariakit** - [Demo](https://codesandbox.io/s/blade-comp-ariakit-composite-ewc9i4?file=/src/App.tsx) | [Doc](https://reakit.io/docs/composite/)
```

Comment:
I think ariakit looks fine to me in this current context based on the flexibility and capabilities it provides. Although documentation can be quite tricky I think if a few basic/advanced interactions are sorted then we shall be ok to take this bet.

Reply:
Yes it's pretty nice :( but the unsuitability is the issue I'm worried about, I don't think we should take this bet. We will be in a cycle of fixing breaking changes.